beanie/odm/documents.py

```diff
@@ -763,7 +763,7 @@ def update_all(
 
     def set(
         self: DocType,
-        expression: Dict[Union[ExpressionField, str], Any],
+        expression: Dict[Any, Any],
```

Comment:
```suggestion
        expression: Dict[Union[ExpressionField, str, Any], Any]
```
Same suggestion for elsewhere too.